### PR TITLE
[MERGE] various: move and reorganize qweb / jinja templates used for mailing

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
PURPOSE

Clean organization of templates in odoo apps: mail.template records in data,
qweb templates (views) used directly in code, notably using post with view.
Purpose is to ease future improvements in posting based on templates.

Remove unused blank templates in crm and mail.

SPECIFICATIONS: TEMPLATES MOVE

  * move those templates in their own file to ease their discovering and
    maintenance;
  * put them into data (as those are not views even if it contains qweb)
  * guidelines are now :

    -> Qweb templates should be in data/mail_templates.xml;
    -> mail.template records should be in data/mail_template_data.xml;

  * put their declaration in no update when not done if template has no
    technical code or complex dependency on underlying code;
  * move found mail data (mail.message.subtype or mail.activity.type) records
    in a mail_data file that should contain only "core" records linked to mail;

SPECIFICATIONS: REMOVE BLANK TEMPLATES

Crm and mail hold a blank template that adds no real value. Its purpose was to
give a base for salesmen to perform a mass mailing on leads. Using a template
and its action menu entry it was possible to launch a mail composer in
mass mail mode on leads.

However template itself is not necessary. Same goal can be achieved with
a standard action. As template was blank and creating noise we can remove
it and simplify available templates.

LINKS

Task ID-2375767
COM PR odoo/odoo#61814
ENT PR odoo/enterprise#14775
UPG PR odoo/upgrade#1936

### CODE REVIEW

File: addons/auth_signup/data/ir_config_parameter_data.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
```
Comment (nseinlet) [inline addons/auth_signup/data/ir_config_parameter_data.xml:3]: I highly suspect this noupdate to be non-working in case of the function tag.

Comment (pedrobaeza) [reply addons/auth_signup/data/ir_config_parameter_data.xml:3]: I think this makes the function to be executed only once (on installation), but no more.

Comment (nseinlet) [reply addons/auth_signup/data/ir_config_parameter_data.xml:3]: this survive an `-u auth_signup` but not a `-i auth_signup`

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by odony** [comment]
@robodoo override=ci/security

**Pull request closed without merging**